packages/pluggableWidgets/rich-text-web/src/RichText.tsx

```diff
@@ -1,79 +1,39 @@
+
+    const onKeyPress = useCallback(() => executeAction(props.onKeyPress), [props.onKeyPress]);
+
     if (props.stringAttribute.status !== "available") {
         return <img src={loadingCircleSvg} className="widget-rich-text-loading-spinner" alt="" aria-hidden />;
     }
-    const onKeyChange = useCallback(() => executeAction(props.onChange), [props.onChange]);
-    const onKeyPress = useCallback(() => executeAction(props.onKeyPress), [props.onKeyPress]);
-    const onChangeFn = useCallback(
-        debounce((value: string) => props.stringAttribute.setValue(value), 500),
-        [props.stringAttribute]
-    );
-    const defineToolbar = (): CKEditorConfig => {
-        const { advancedConfig, toolbarConfig, preset } = props;
-        if (preset !== "custom") {
-            return getPreset(preset);
-        } else {
-            const groupKeys = Object.keys(props).filter((key: string) => (key.includes("Group") ? key : null));
-            let groupItems: ToolbarItems[] | string[];
-            if (toolbarConfig === "basic") {
-                groupItems = groupKeys
-                    .filter((groupName: GroupType) => props[groupName])
-                    .map((groupName: GroupType) =>
-                        groupName.includes("separator") ? "/" : groupName.replace("Group", "").toLowerCase()
-                    );
-            } else {
-                groupItems = defineAdvancedGroups(advancedConfig);
-            }
 
-            return createCustomToolbar(groupItems, toolbarConfig === "basic");
-        }
-    };
-    const plugins = [];
-    if (props.codeHighlight) {
-        plugins.push("codesnippet");
-    }
     return (
         <RichTextComponent
-            advancedConfig={props.advancedConfig}
-            advancedContentFilter={
-                props.advancedContentFilter === "custom"
-                    ? {
-                          allowedContent: props.allowedContent,
-                          disallowedContent: props.disallowedContent
-                      }
-                    : null
-            }
-            sanitizeContent={props.sanitizeContent}
-            name={props.name}
-            editorType={props.editorType}
-            readOnlyStyle={props.readOnlyStyle}
-            readOnly={props.stringAttribute.readOnly}
-            enterMode={props.enterMode}
-            shiftEnterMode={props.shiftEnterMode}
-            spellChecker={props.spellChecker}
-            toolbar={defineToolbar()}
-            plugins={plugins}
-            value={props.stringAttribute.value}
-            onValueChange={onChangeFn}
+            key={editorKey}
+            onChange={onChange}
             onKeyPress={onKeyPress}
-            onKeyChange={onKeyChange}
-            dimensions={{
-                width: props.width,
-                widthUnit: props.widthUnit,
-                height: props.height,
-                heightUnit: props.heightUnit
-            }}
-            tabIndex={props.tabIndex}
```

Comment:
It means now we do not pass `name` and `tabIndex` to the widget anymore? Also, why it was necessary to rebuild the whole widget?